python/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -279,17 +270,19 @@ class KNeighborsClassifier(NearestNeighbors):
             When set to True, the fit method will automatically
             convert the inputs to np.float32.
         """
+
+        out_type = self._get_output_type(X)
```

Comment:
Same as above- use an instance variable for this so the user has an option to match the input type or set an explicit output type.